### ISSUE TITLE
global: url decoding check

### ISSUE THREAD

**Issue by lnielsen** [inveniosoftware/invenio-oauth2server#123]
Check if oauthlib's strict URL decoding is enforced. See #47

- oauthlib have strict decode
- invenio-oauh2server allows you to override that strict configuration

1) Check if invenio-search-js is properly sending fully encoding URL strings so that we don't get internal server errors.

**Comment by lnielsen** [comment]
JS part fixed in inveniosoftware/invenio-search-js#96 

Also we need to add an error handler that catches the URL decoding error and replies with a 400 error so that they know they need to properly URL-encode their request. Currently they get a 500 error.

Right now this is the stack trace:

```python
Stacktrace (most recent call last):

  File "flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "flask/app.py", line 1637, in full_dispatch_request
    rv = self.preprocess_request()
  File "flask/app.py", line 1837, in preprocess_request
    rv = func()
  File "invenio_oauth2server/ext.py", line 171, in verify_oauth_token_and_set_current_user
    valid, req = oauth2.verify_request(scopes)
  File "flask_oauthlib/provider/oauth2.py", line 483, in verify_request
    uri, http_method, body, headers, scopes
  File "oauthlib/oauth2/rfc6749/endpoints/base.py", line 64, in wrapper
    return f(endpoint, uri, *args, **kwargs)
  File "oauthlib/oauth2/rfc6749/endpoints/resource.py", line 68, in verify_request
    request = Request(uri, http_method, body, headers)
  File "oauthlib/common.py", line 405, in __init__
    self._params.update(dict(urldecode(self.uri_query)))
  File "oauthlib/common.py", line 131, in urldecode
    raise ValueError(error % (set(query) - urlencoded, query))
```

**Comment by drjova** [comment]
@lnielsen this specific stack trace comes from ``verify_oauth_token_and_set_current_user`` and is solved here https://github.com/inveniosoftware/invenio-oauth2server/pull/135
